site/blog/unlisted/iasql-on-ec2.mdx

```diff
@@ -208,3 +200,130 @@ WHERE id = (SELECT id
+
+#### Import Domain In Route53
+
+Let's start with adding our domain to Route53. To be able to manage Route53 resources in IaSQL, we'll first need to install the corresponding module:
+```sql
+SELECT iasql_install('aws_route53');
+```
+Alright, now let's `INSERT` a row into the `hosted_zone` table to direct IaSQL engine to create a new Route53 hosted zone for us.
+
+:::note
+To know which tables are supported by the `aws_route53` module, you can visit [`aws_route53` module documentation](https://iasql.com/docs/modules/aws/aws_route53/).
+:::
+
+```sql
+SELECT iasql_begin();
+
+INSERT INTO hosted_zone (domain_name)
+VALUES ('<your-domain>');
+
+SELECT iasql_commit();
+```
+
+:::note
+Simple `INSERT`, right? But wait! What are those `iasql_begin` and `iasql_commit` functions called before and after our insert command?
+
+Quick answer is that if you need your changes to be applied now instead of eventually in the future, you'll wrap the changes between an `iasql_begin` and an `iasql_commit`. For more info on what they do, visit [this page](https://iasql.com/docs/transaction/).
+:::
+
+Now that we have created our hosted zone, we need to set our domain's nameservers to the ones that Route53 provides us. For each public hosted zone that you create, Amazon Route53 [automatically creates](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/SOA-NSrecords.html) a name server (NS) record and a start of authority (SOA) record. So we should log into the domain registrar, and set the name server values to the four name server values that Route53 has set. To get those values, we can simply run a `SELECT` query:
+```sql
+SELECT record as name_server
+FROM resource_record_set
+WHERE parent_hosted_zone_id = (SELECT id FROM hosted_zone WHERE domain_name = '<your-domain>')
+  AND record_type = 'NS';
+```
+After updating the name servers in the domain settings, we need to wait for those values to be propagated. This can happen in minutes, or even up to 48 hours depending on the registrar.
+
+#### Issue An SSL Certificate For Our Domain
+Having the domain setup in Route53, it's easy to use IaSQL's `certificate_request` RPC to issue a new certificate for it. If you try to go with this procedure through AWS console, you'll need to manually add the CNAME records from the AWS ACM to your hosted zone. But this RPC is implemented in a way that automagically [sets the DNS records](https://github.com/iasql/iasql/blob/v0.1.7/src/modules/aws_acm/rpcs/request.ts#L177-L188), [waits for the certificate to be issued](https://github.com/iasql/iasql/blob/v0.1.7/src/modules/aws_acm/rpcs/request.ts#L197), and then [removes those validation DNS records](https://github.com/iasql/iasql/blob/v0.1.7/src/modules/aws_acm/rpcs/request.ts#L200).
+
+```sql
+-- SELECT iasql_install('aws_acm'); -- no need for this because aws_route53 depends on it and it's automatically installed
+SELECT certificate_request('<your-domain>', 'DNS', default_aws_region(), '{}');
```

Comment:
Note for a future feature addition: there should be a simplification of this function to not need the last 2 (or even 3?) arguments. These seem like reasonable defaults to me.

Reply:
agreed